### ISSUE TITLE
Same code produces different results in VScode vs terminal

### ISSUE THREAD

**Issue by fabrizioleone** [robertfeldt/BlackBoxOptim.jl#220]
Hi, 

Thanks for this great package.

I am using ```bboptimize``` to find the minimum of a non-linear function. My approach is very simple

```Julia
res = bboptimize(x -> my_obj_fun(x, args); SearchRange = [(0.0, 2.0), (0.0, 2.0)], Method = :adaptive_de_rand_1_bin)
```

This code produces different results if I run it on VScode or if I call Julia in batch mode from the terminal of my computer. Since everything else is identical (package version, seeds, and objective function) I wonder if ```bboptimize``` searches differently depending on how it is called.

Is this a known issue? Is there any way to make sure ```bboptimize``` gives the same results in both cases?

Thank you.

**Comment by fabrizioleone** [comment]
Hi, thanks. I did set the same seeds already. So I don’t think that’s the issue.

On 15 Aug 2023, at 10:58, Robert Feldt ***@***.***> wrote:

﻿

Try to set the same random seed on both platforms otherwise the results might differ between runs since these algorithms are stochastic.

**Comment by robertfeldt** [comment]
Hmm, then I don't understand it. If you're on the same machine, same Julia version, and everything else is the same there shouldn't be a difference. Are you sure VSCode is invoking the same Julia binary as when you run from the terminal? Maybe include a call to ```versioninfo()```right before the call to bboptimize? What do you mean "searches differently depending on how it is called"? There is no difference in the code, you mean "from where" it is called, right?

**Comment by JobJob** [comment]
Maybe `init_rng!(params)` should be called in `bbsetup`? Populations get initialized there (which use the PRNG), i.e. seems it should be called before `optimizer = optimizer_func(problem, params)` [here](https://github.com/robertfeldt/BlackBoxOptim.jl/blob/37d1317a62e0a4bc4053b89599f2323c44ae036b/src/bboptimize.jl#L103)

**Comment by robertfeldt** [comment]
@JobJob but why would we need to do this within BlackBoxOptim when the user can do it themselves if they need a particular RNG "behavior"? Please clarify, maybe I'm missing something.

**Comment by JobJob** [comment]
Just if you wanted the same seed to produce the same result, for reproducibility. This is what I assumed would happen if I called `bbsetup(f; ..., RandomizeRngSeed=false, RngSeed=42)` that I'd get the same result every time, but I don't because of the setup that happens (initial population creation) before the RNG is seeded in `init_rng`

**Comment by JobJob** [comment]
I guess my thinking is: what's the point of providing a RngSeed if it doesn't lead to reproducible results?

**Comment by robertfeldt** [comment]
Yes, for sure, you have a point. I'm considering skipping the fiddling with the RNG inside of the package though. Hmm, let me revisit the design around this, not sure why we introduced that in the first place... ;)

**Comment by JobJob** [comment]
Yeah I think that makes sense to remove all the RNG seeding stuff completely.

________________________________

**Comment by robertfeldt** [comment]
Ok, now changed on master:

https://github.com/robertfeldt/BlackBoxOptim.jl/commit/f3b8a2b0a7bbc93a690c615b6905b501e85de97c
